### PR TITLE
docs: fix leader action transition descriptions

### PR DESCRIPTION
**Pull request by jrudolph**
In https://github.com/akka/akka/pull/28293/files#r353183941 I was confused about whether the current description is right and I still think it is wrong.

According to the same page:

> Joining members will be promoted to WeaklyUp and become part of the cluster if convergence can’t be reached. Once gossip convergence is reached, the leader will move WeaklyUp members to Up.

Weakly Up -> Up needs convergence like all other leader actions.

The question is still how the edge from Joining -> WeaklyUp is done. The above paragraph uses passive voice to dodge that question but the code doesn't lie:

https://github.com/akka/akka/blob/d0027850e6ab2c229c1eaa07e1ce872ca14113e1/akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala#L1146-L1182

So, this transition is also a leader action but it doesn't require convergence.

Did I get this wrong?

### CODE REVIEW

**Comment by johanandren** [comment]
No that's technically correct but the page makes it easy for itself by basically saying `leader action` are synonymous with requiring convergence, vs explaining the whole affair of there possible can be multiple leader in the face of a partition, rather than changing that and rewording other sections I thought it'd be better to just go along with that.

I think fixing it requires more changes to the page and I am not convinced it is worth it.

**Review by patriknw — APPROVED** [review]
Yes, it's performed by the leader (smallest address among the reachable) and could even be done by several such leaders, one in each partition. Convergence not required for that action (that is the point of having WeaklyUp).

**Comment by johanandren** [comment]
To be more concrete, for these two sections need rewording and expanding:

https://github.com/akka/akka/pull/28306/files#diff-606795fe54009ad5b3887b2a09d6f34fL106
https://github.com/akka/akka/pull/28306/files#diff-606795fe54009ad5b3887b2a09d6f34fL74

**Comment by helena** [comment]
> To be more concrete, these two sections need rewording and expanding:
> 
> https://github.com/akka/akka/pull/28306/files#diff-606795fe54009ad5b3887b2a09d6f34fL106
> https://github.com/akka/akka/pull/28306/files#diff-606795fe54009ad5b3887b2a09d6f34fL74

+1 it makes sense that if we clarify the initial proposal, which seems correct, that these to raised by @johanandren also need a clarification, to not be further misleading.

**Comment by patriknw** [comment]
@jrudolph can you complete this so we get it out of the way?

**Comment by jrudolph** [comment]
> @jrudolph can you complete this so we get it out of the way?

Yep, I'll have a look.

File: akka-docs/src/main/paradox/typed/cluster-membership.md [hunk 1]
```diff
@@ -58,60 +58,74 @@ Note that the node might already have been shutdown when this event is published
...
 
 If a node is `unreachable` then gossip convergence is not possible and therefore
-any `leader` actions are also not possible (for instance, allowing a node to
-become a part of the cluster). To be able to move forward the state of the
-`unreachable` nodes must be changed. It must become `reachable` again or marked
-as `down`. If the node is to join the cluster again the actor system must be
-restarted and go through the joining process again. If new incarnation of the unreachable
-node tries to rejoin the cluster old incarnation will be marked as `down` and new
-incarnation can rejoin the cluster without manual intervention. 
+most `leader` actions are impossible (for instance, allowing a node to
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/typed/cluster-membership.md]: I added a bit more explanations here about the reasoning about why things are the way they are.

File: akka-docs/src/main/paradox/typed/cluster-membership.md [hunk 2]
```diff
@@ -58,60 +58,74 @@ Note that the node might already have been shutdown when this event is published
...
+unreachable and then regains connectivity cannot rejoin the cluster. Instead, the process has to be restarted on the
+node, creating a new actor system that can go through the joining process again.
+
+A special case is a node that was restarted without going through the leaving or downing process e.g. because the machine
+hosting the node was unexpectedly restarted. When the new instance of the node tries to rejoin the cluster, the cluster
+might still track the old instance as unreachable. In this case, however, it is clear that the old node is gone because
+the new instance will have the same address (host and port) as its old instance. In this case, the previous instance
+will be automatically marked as `down` and the new instance can rejoin the cluster without manual intervention.
+
+## Leader
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/typed/cluster-membership.md]: I added this complete section to explain the leader and how it relates to state changes.

File: akka-docs/src/main/paradox/typed/cluster-membership.md [hunk 3]
```diff
@@ -58,60 +58,74 @@ Note that the node might already have been shutdown when this event is published
...
-convergence can't be reached. Once gossip convergence is reached, the leader will move `WeaklyUp`
-members to `Up`.
-
-This feature is enabled by default, but it can be disabled with configuration option:
-
-```
-akka.cluster.allow-weakly-up-members = off
-```
-
-You can subscribe to the `WeaklyUp` membership event to make use of the members that are
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/typed/cluster-membership.md]: This looks like a duplication left over from a merge? I didn't dig into what was supposed to be the ultimate result of the merge but mostly edited the text to make sense.

**Review by johanandren — APPROVED** [review]
Great!

**Review by patriknw — APPROVED** [review]
LGTM again, reviewed the latest

**Pull request merged**